src/compile/layout/header.ts

```diff
@@ -97,6 +97,30 @@ export function getHeaderGroups(model: Model, channel: HeaderChannel): VgMarkGro
   return groups;
 }
 
+// 0, (0,90), 90, (90, 180), 180, (180, 270), 270, (270, 0)
+
+export function labelAlign(angle: number) {
+  // to keep angle in [0, 360)
+  angle = ((angle % 360) + 360) % 360;
+  if ((angle + 90) % 180 === 0) {  // for 90 and 270
+    return {}; // default center
+  } else if (angle < 90 || 270 < angle) {
+    return {align: {value: 'right'}};
+  } else if (135 <= angle && angle < 225) {
+    return {align: {value: 'left'}};
+  }
+  return {};
+}
+
+export function labelBaseline(angle: number) {
+  // to keep angle in [0, 360)
+  angle = ((angle % 360) + 360) % 360;
+  if (80 <= angle && angle <= 135) {
```

Comment:
80?

Reply:
I chose it because I felt it looked good. What do you think should be the rule for this?

Reply:
Didn't we look at the rules together? I don't remember 80 coming up anywhere.

Reply:
All the angle we use elsewhere is divisible by 45.  We should make it consistent.

Reply:
We didn't look at header rules. We only looked at axis labels, @domoritz.